app/scripts/lib/stores/read-only-network-store.test.ts

```diff
@@ -0,0 +1,141 @@
+ */
+function setupReadOnlyNetworkStore() {
+  const store = new ReadOnlyNetworkStore();
+  return store;
+}
+
+/**
+ * Create a Nock scope for the fixture server response.
+ *
+ * @returns A Nock interceptor for the fixture server response.
+ */
+function mockFixtureServerInterceptor(): nock.Interceptor {
+  return nock(FIXTURE_SERVER_ORIGIN).get(FIXTURE_SERVER_PATH);
+}
+
+/**
+ * Create a Nock scope for the fixture server response, which will have a successful reply.
+ *
+ * @param state
+ */
+function setMockFixtureServerReply(
+  state: Record<string, unknown> = DEFAULT_INITIAL_STATE,
+): void {
+  mockFixtureServerInterceptor().reply(200, state);
+}
+
+describe('ReadOnlyNetworkStore', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    nock.cleanAll();
+  });
+
+  describe('constructor', () => {
+    it('makes a request to the fixture server upon construction', async () => {
+      const scope = nock(FIXTURE_SERVER_ORIGIN)
+        .get(FIXTURE_SERVER_PATH)
+        .reply(200, 'Test');
+      expect(scope.pendingMocks().length).toBe(1);
+      setupReadOnlyNetworkStore();
+      await new Promise(process.nextTick);
+      expect(scope.pendingMocks().length).toBe(0);
+    });
+  });
+
+  describe('initialization (#init)', () => {
+    it('loads state from the network if fetch is successful and response is ok', async () => {
+      setMockFixtureServerReply(MOCK_STATE);
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toStrictEqual(MOCK_STATE);
+    });
+
+    it('does not throw, and logs a debug message, if fetch is not okay', async () => {
+      const logDebugSpy = jest
+        .spyOn(log, 'debug')
+        .mockImplementation(() => undefined);
+
+      mockFixtureServerInterceptor().reply(400);
+
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toBe(null);
+      expect(logDebugSpy).toHaveBeenCalledWith(
+        'Received response with a status of 400 Bad Request',
+      );
+    });
+
+    it('does not throw, and logs a debug message, if fetch throws an error', async () => {
+      mockFixtureServerInterceptor().replyWithError('error!');
+      const logDebugSpy = jest
+        .spyOn(log, 'debug')
+        .mockImplementation(() => undefined);
+
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toBe(null);
+      expect(logDebugSpy).toHaveBeenCalledWith(
+        "Error loading network state: 'request to http://localhost:12345/state.json failed, reason: error!'",
+      );
+    });
+  });
+
+  describe('get', () => {
+    it('returns null if #state is null', async () => {
+      mockFixtureServerInterceptor().reply(200);
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toBe(null);
+    });
+
+    it('returns stored state if #state has data', async () => {
```

Comment:
Nit: Avoid referencing internal implementation details in test descriptions
```suggestion
    it('returns null if state is null', async () => {
      mockFixtureServerInterceptor().reply(200);
      const store = setupReadOnlyNetworkStore();
      const result = await store.get();
      expect(result).toBe(null);
    });

    it('returns stored state if state has data', async () => {
```

Reply:
suggestion applied in 23bb69440e